socs/testing/hwp_emulator.py

```diff
@@ -0,0 +1,223 @@
+"""
+HWP Emulation module
+"""
+import logging
+import threading
+import time
+from dataclasses import dataclass
+
+import pytest
+
+from socs.agents.hwp_pid.drivers.pid_controller import PID
+from socs.testing import device_emulator
+
+
+def hex_str_to_dec(hex_value, decimal=3):
+    """Converts a hex string to a decimal float"""
+    return float(int(hex_value, 16)) / 10**decimal
+
+
+@dataclass
+class PMXState:
+    """State of the PMX Emulator"""
+    output: bool = False
+    current: float = 0
+    current_limit: float = 10.0
+    voltage_limit: float = 10.0
+    voltage: float = 0
+    source: str = "volt"
+
+
+@dataclass
+class PIDState:
+    """State of the PID Emulator"""
+    direction: str = "forward"
+    freq_setpoint: float = 0.0
+
+
+@dataclass
+class HWPState:
+    """State of the HWP Emulator"""
+    cur_freq: float = 0.0
+    pmx: PMXState = PMXState()
+    pid: PIDState = PIDState()
```

Comment:
That is a good catch, that must be new to python3.11 since I'm running 3.10 and don't get those errors.  I don't feel the need to upgrade my system python right this moment but will probably do so pretty soon. I'll fix those defaults though